lib/workers/repository/update/pr/index.ts

```diff
@@ -73,6 +77,27 @@ export function updatePrDebugData(
   };
 }
 
+function validatePrCache(
+  branchName: string,
+  prCache: PrCache,
+  prFingerprint: string
+): boolean {
+  if (prCache.fingerprint !== prFingerprint) {
+    logger.debug('PR fingerprints mismatch, processing PR');
+    setPrCache(branchName, prFingerprint);
```

Comment:
Shouldn't this be set *after* it's been updated? i.e. in case update is skipped or aborted for some other reason

Reply:
You're right. 
For the prCache to be set after updates it is necessary to use filtered config, else we will not be in sync in some cases.
For eg. 
When a new PR is created some config fields e. `updatedPackageFiles` are populated, while on the next run the these fields are empty. This results in diff fingerprint in between runs but the prBody doesn't depend on these fields this the oldPrBodyHash is same as newPrBodyHash and the updates will be skipped & the prCache   doesn;t get updated.

So I will add this logic along with a function to generate prFingeprintConfig and defined a PrFingerprintConfig.